src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6642,46 +6689,53 @@ private ImmutableArray<FlowAnalysisAnnotations> GetParametersAnnotations(Immutab
         }
 
         private VisitArgumentResult VisitArgumentEvaluate(BoundExpression argument, RefKind refKind, FlowAnalysisAnnotations annotations)
+        {
+            var savedState = VisitArgumentEvaluatePrologue(argument);
+            Visit(argument);
+            return VisitArgumentEvaluateEpilogue(argument, savedState, refKind, annotations);
+        }
+
+        private Optional<LocalState> VisitArgumentEvaluatePrologue(BoundExpression argument)
         {
             Debug.Assert(!IsConditionalState);
-            var savedState = (argument.Kind == BoundKind.Lambda) ? this.State.Clone() : default(Optional<LocalState>);
+            return (argument.Kind == BoundKind.Lambda) ? this.State.Clone() : default(Optional<LocalState>);
+        }
+
+        private VisitArgumentResult VisitArgumentEvaluateEpilogue(BoundExpression argument, Optional<LocalState> savedState, RefKind refKind, FlowAnalysisAnnotations annotations)
+        {
             // Note: DoesNotReturnIf is ineffective on ref/out parameters
 
             switch (refKind)
```

Comment:
>switch (refKind) [](http://example.com/codeflow?start=12&length=16)

Consider adding ```default:``` case that throws (since we are "extracting" ```Visit``` out of this switch).